### ISSUE TITLE
Missing details for data preprocessing

### ISSUE THREAD

**Issue by Treeboy2762** [MIC-DKFZ/VoxTell#4]
Dear authors,

Thank you for releasing the model weights, and congratulations again for your progress with the project.

I wanted to replicate the training of VoxTell for RexGroundingCT, which involves chest CT images from CT-RATE. However, the CT images have variable voxel spacing and 1mm spacing won't allow images to fit in 192 cube voxels. Did you resample volumes to a fixed target spacing during training, and use sliding windows inference?

Best,
Daesung

**Comment by moritzlangenberg** [comment]
Dear Daesung,

Thank you for your interest and for reaching out!

Both training and inference of VoxTell are performed patch-based, with a patch size of 192x192x192. During inference, we use sliding-window prediction. VoxTell is trained without resampling, i.e., images are processed at their native resolution.
Hope this helps, feel free to ask if anything is unclear.

Cheers,
Moritz

**Comment by Treeboy2762** [comment]
Hi Moritz,

Thank you so much for the clarification!! As a follow-up, I have performed a quick visualization of the dimensions for ReXGroundingCT, stratified by the split:

<img width="1784" height="784" alt="Image" src="https://github.com/user-attachments/assets/95ad82fe-75ca-49ec-8187-ca81a5d35b3e" />

I noticed some variability in the voxel spacing (Q1-Q3: [0.65, 0.75] for x and y, [0.75, 1.25] for z), and I was curious about the reason behind processing at native resolution.

1) Would the network be robust to different voxel spacing? (same performance over different voxel spacing?)
2) Were there special mining techniques to mitigate oversampling of negative patches? In big scans (>= 512 512 in x, y) I believe that it is likely to sample more patches without annotations..

I really appreciate your time :) Happy new year!

Best,
Daesung

**Comment by moritzlangenberg** [comment]
Hi Daesung,

You’re welcome!

Regarding 1., VoxTell’s training data contains a large variety of datasets with different spacings, and we use nnUNet data augmentation, leading to robustness to different voxel spacings. For images with very high resolution or out-of-distribution spacings, we suggest downsampling to a common standard resolution.

Regarding 2., we oversample foreground patches with a probability of 85%, and within those patches we again sample 75% foreground labels (e.g., 3 of 4 text prompts occur in the patch).

I hope this clarifies your questions. Wishing you a happy new year as well!

Cheers,
Moritz

**Comment by Treeboy2762** [comment]
Thank you so much Moritz! My concerns have been cleared.

If you may, is the final checkbox in the roadmap (Support and scripts for custom fine-tuning) meant for integration to the nnUNet pipeline?? I am curious if you would release the training scripts as well, which I assume is nnUNet-based.
